scripts/find_duplicates.R

```diff
@@ -0,0 +1,35 @@
+library("magrittr")
+
+get_links <- function(path){
+  path %>%
+    readLines(warn = FALSE) %>%
+    .[1:which(grepl("Upcoming Events", .))] %>%
+    commonmark::markdown_xml() %>%
+    xml2::read_xml() %>%
+    xml2::xml_ns_strip() %>%
+    xml2::xml_find_all(xpath = './/link') %>%
+    xml2::xml_attr("destination")
+}
+
+get_dups <- function(rweekly_path = getwd()){
+  old_posts <- fs::dir_ls(file.path(
+    rweekly_path,"_posts"))
+  
+  old_posts <- old_posts[(length(old_posts) - 20):length(old_posts)]
+  
+  
+  old_links <- unlist(purrr::map(old_posts, get_links))
+  
+  thisweek <- get_links(file.path(rweekly_path,
+                                  "draft.md"))
+  dups <- thisweek[thisweek %in% old_links]
```

Comment:
What about also checking the current week for duplicates - I'm always concerned I'm going to add something twice.

```suggestion
  current_dups <- thisweek[duplicated(thisweek)]
  if (length(current_dups)) {
    message("Duplicates found in current week:")
    cat(paste0(current_dups, collapse = "\n"))
    cat("\n")
  }
  
  dups <- thisweek[thisweek %in% old_links]
```

Reply:
why don't I ever use `setdiff()`?!